src/presenter/events-presenter.js

```diff
@@ -67,19 +69,29 @@ export default class EventListPresenter {
     render(new AddEventView(), this.#eventListComponent.element);
   }
 
+  #renderSort(){
+    render(new SortView(SORT_ITEMS), this.#container);
+  }
+
   #renderTrip() {
+    this.#renderSort();
     this.#renderEventList();
     for (const event of this.events){
       const destination = this.#eventsModel.getDestination(event.destination);
       const offersInfo = event.offers.map((element) => this.#eventsModel.getOffer(event.type, element));
-      this.#renderEvent(event, destination, offersInfo, this.cities);
+      const offers = this.#eventsModel.getOffers(event.type);
+      this.#renderEvent(event, destination, offersInfo, this.cities, offers);
     }
   }
 
   init() {
     this.events = this.#eventsModel.events;
-    this.destinations = this.#eventsModel.destinations;
-    this.cities = this.#eventsModel.cities;
-    this.#renderTrip();
+    if (this.events.length > 0) {
+      this.destinations = this.#eventsModel.destinations;
+      this.cities = this.#eventsModel.cities;
+      this.#renderTrip();
+    } else {
+      render(new EmptyListView(), this.#container);
+    }
```

Comment:
Можно снизить вложенность и избавиться от `else`

```suggestion
    if (!this.events.length) {
      render(new EmptyListView(), this.#container);
      return;
    }
    
    this.destinations = this.#eventsModel.destinations;
    this.cities = this.#eventsModel.cities;
    this.#renderTrip();    
```